internal/dcache/rpc/server/handler.go

```diff
@@ -47,43 +57,980 @@ var _ service.ChunkService = &ChunkServiceHandler{}
+
+	// count of in-progress chunk operations (get, put or remove) for this MV.
+	// This is used to block the end sync call till all the ongoing chunk operations are completed.
+	chunkIOInProgress atomic.Int64
+
+	// Two MV states are interesting from an IO standpoint.
+	// An online MV is the happy case where all RVs are online and sync'ed. In this state there won't be any
+	// resync Writes, and client Writes if any will be replicated to all the RVs, each of them storing the chunks
+	// in their respective mv folders. This is the normal case.
+	// A syncing MV is interesting. In this case there are resync writes going and possibly client writes too.
+	// Client chunks are saved in a special mv.sync folder while sync writes are saved in the regular mv folder.
+	// During EndSync, the client writes are moved from mv.sync to the regular mv folder and then we have the MV
+	// back in online state.
+	// The short period when an MV moves in and out of syncing state is important. We need to quiesce any IOs
+	// to make sure we don't miss resyncing any chunk.
+	// Both StartSync and EndSync will quiesce IOs just before they move the mv into and out of syncing state, and
+	// resume IOs once the MV is safely moved into the new state.
+	//
+	// This boolean flag will be set by StartSync and EndSync to put the mv in quiesce state and it'll be read
+	// and honored by the various chunk related APIs - GetChunk, PutChunk, etc.
+	quiesceIOs atomic.Bool
+
+	syncInfo // sync info for this MV
+}
+
+type syncInfo struct {
+	// Is the MV in syncing state?
+	// An MV enters syncing state after StartSync command is successfully executed.
+	// In syncing state, PutChunk requests corresponding to client writes will be
+	// saved in the mv#.sync folder. Similarly a successful EndSync will take an
+	// MV out of syncing state.
+	isSyncing atomic.Bool
+
+	// sync ID for this MV if it is in syncing state.
+	// This is returned in the StartSync response and EndSync should carry this.
+	syncID string
+
+	// sourceRV is the RV that is syncing a (local) target RV in this MV.
+	// Since not more than one component RVs of an MV will be from the same node, there will be one and only
+	// one local RV as the target RV.
+	// Communicated using the StartSync message sent by the Replication Manager as part of the resync-mv workflow.
+	// We will only accept PutChunk(isSync=true) requests only with source RV value matching this.
+	// This is used when the source RV goes offline and is replaced by the cluster manager with a new RV.
+	// In this case, the sync will need to be restarted from the new RV.
+	sourceRVName string // source RV name for syncing this MV
 }
 
 func NewChunkServiceHandler() *ChunkServiceHandler {
+	// TODO:: integration: get rvID, rvName and cache dir path for different RVs for the node from cluster manager
+	// below will be call to cluster manager to get the information
+	rvIDMap := getRvIDMap()
+
```

Comment:
we can have cluster manager call the start method of this server, passing
(dCacheConfig *dcache.DCacheConfig, rvs []dcache.RawVolume)

Reply:
Will do in next PR